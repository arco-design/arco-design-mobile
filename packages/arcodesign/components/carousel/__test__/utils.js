import { act } from 'react-dom/test-utils';
import { createMoveTouchEventObject, createStartTouchEventObject } from '../../../tests/helpers/mockEvent';

export function mockSwipeStart(map, clientXMap) {
    const { touchstart, touchmove } = clientXMap;
    act(() => {
        map.touchstart(createStartTouchEventObject({ x: touchstart, y: 0 }));
        map.touchmove(createMoveTouchEventObject({ x: touchmove, y: 0 }));
    });
}
export function mockSwipeEnd(wrapper, className, touchend) {
    wrapper.find(`.${className}`).simulate('touchEnd', createMoveTouchEventObject({ x: touchend, y: 0 }));
    act(() => {
        jest.advanceTimersByTime(600);
    });
    wrapper.update();
}

export function mockSwipe(map, wrapper, className, clientXMap) {
    const { touchend } = clientXMap;
    mockSwipeStart(map, clientXMap);
    mockSwipeEnd(wrapper, className, touchend);
}
