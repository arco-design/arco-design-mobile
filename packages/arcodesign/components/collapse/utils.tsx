import React, { Context } from 'react';
import { GroupContextParams } from './type';

export const defaultContextKey = '__COLLAPSE_CONTEXT__';
export const allContexts: Record<string, Context<GroupContextParams>> = {};

export function getCollapseContext(key: string) {
    if (!allContexts[key]) {
        allContexts[key] = React.createContext<GroupContextParams>({
            isGroup: false,
            value: [],
            onCollapse: () => {},
        });
    }
    return allContexts[key];
}

export const CollapseKeyContext = React.createContext<{ key: string }>({
    key: defaultContextKey,
});
