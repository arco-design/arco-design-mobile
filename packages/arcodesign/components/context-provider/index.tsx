import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
