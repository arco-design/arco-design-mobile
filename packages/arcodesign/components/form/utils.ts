import { IRules, ValidatorError } from '@arco-design/mobile-utils';

export const isFieldRequired = (rules: IRules[] = []) => {
    return (rules || []).some(rule => rule?.required);
};

export const getErrorAndWarnings = (result: ValidatorError[]) => {
    let errors: string[] = [];
    let warnings: string[] = [];
    result.map(({ message = [], validateLevel = 'error' }) => {
        if (!message?.length) {
            return;
        }
        if (validateLevel === 'warning') {
            warnings = [...warnings, ...message];
        } else {
            errors = [...errors, ...message];
        }
    });
    return { warnings, errors };
};
