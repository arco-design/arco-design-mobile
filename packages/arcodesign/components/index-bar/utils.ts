import { Children, isValidElement, ReactElement, ReactNode } from 'react';
import { IndexBarIndexType } from './type';

export function filterValidIndexBarChild(children: ReactNode) {
    const validChildren: ReactElement[] = [];
    Children.forEach(children, child => {
        if (!isValidElement(child)) {
            console.warn('<IndexBar /> only accept <IndexBar.Group /> as the children');
            return;
        }
        validChildren.push(child);
    });
    return validChildren;
}

// index有可能是number或者string，但是绑在data-index的attr上后
// 通过滑动事件得到的都是string，需要处理下
export function getFormatIndex(
    index: IndexBarIndexType,
    activeIndex: IndexBarIndexType | undefined,
) {
    const activeIndexType = typeof activeIndex;
    const indexType = typeof index;
    if (indexType === activeIndexType) {
        return index;
    }
    if (activeIndexType === 'number') {
        return Number(index);
    }
    return index.toString();
}

export function isValidIndex(index: unknown): index is IndexBarIndexType {
    return typeof index === 'number' || typeof index === 'string';
}

export function getGroupDomFromIndex(containerDom: HTMLDivElement, activeIndex: IndexBarIndexType) {
    let targetDom: HTMLDivElement | null = null;

    const containerChildren = containerDom.children;
    for (let i = 0; i < containerChildren.length; i++) {
        const dom = containerChildren[i] as HTMLDivElement;
        const dataset = dom?.dataset;
        if (dataset && dataset.index) {
            const formatIndex = getFormatIndex(dataset.index, activeIndex);
            if (formatIndex === activeIndex) {
                targetDom = dom;
                break;
            }
        }
    }

    return targetDom;
}
