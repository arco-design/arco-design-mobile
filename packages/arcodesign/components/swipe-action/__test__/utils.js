import { act, fireEvent } from '@testing-library/react';
import {
    createMoveTouchEventObject,
    createStartTouchEventObject,
} from '../../../tests/helpers/mockEvent';

export function mockSwipeStart(comp, clientXMap) {
    const { touchstart, touchmove } = clientXMap;
    act(() => {
        fireEvent.touchStart(
            comp,
            createStartTouchEventObject({ x: touchstart, px: touchstart, y: 0 }),
        );
        fireEvent.touchMove(
            comp,
            createMoveTouchEventObject({ x: touchmove, px: touchmove, y: 0 }),
        );
    });
}
export function mockSwipeEnd(comp, touchend) {
    fireEvent.touchEnd(comp, createMoveTouchEventObject({ x: touchend, px: touchend, y: 0 }));
    act(() => {
        jest.advanceTimersByTime(600);
    });
}

export function mockSwipe(comp, clientXMap) {
    const { touchend } = clientXMap;
    mockSwipeStart(comp, clientXMap);
    mockSwipeEnd(comp, touchend);
}
