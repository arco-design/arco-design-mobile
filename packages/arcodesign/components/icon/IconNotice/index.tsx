import React, { SVGAttributes, CSSProperties } from 'react';
import { ContextLayout } from '../../context-provider';

export interface IconProps extends SVGAttributes<SVGElement> {
    className?: string;
    useCurrentColor?: boolean;
    style?: CSSProperties;
}

export default function IconNotice(props: IconProps) {
    const { className = '', useCurrentColor = true, style, ...other } = props;

    return (
        <ContextLayout>
            {({ prefixCls }) => (
                <svg
                    className={`${prefixCls}-icon ${prefixCls}-icon-notice ${className}`}
                    width="1em"
                    height="1em"
                    style={style}
                    fill={useCurrentColor ? 'currentColor' : '#000'}
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                    {...other}
                >
                    <defs />
                    <path d="M618.667 853.333A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248V704h64a21.333 21.333 0 0121.334 21.333V768a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 768v-42.667A21.333 21.333 0 01106.667 704h64V459.008c0-164.181 108.8-302.123 256-341.248V85.333A21.333 21.333 0 01448 64h128zm-64 128c-141.376 0-256 119.595-256 267.136V704h512V459.136C768 311.595 653.376 192 512 192z" />
                </svg>
            )}
        </ContextLayout>
    );
}
