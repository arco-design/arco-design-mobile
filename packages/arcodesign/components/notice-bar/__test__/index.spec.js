import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import NoticeBar from '..';
import IconNotice from '../../icon/IconNotice';
import { defaultContext } from '../../context-provider';
import { defineHtmlRefProperties } from '../../../tests/helpers/mockElement';
import { pureDelay } from '../../../tests/helpers/utils';

const { prefixCls } = defaultContext;
const prefix = `${prefixCls}-notice-bar`;

demoTest('notice-bar');

mountTest(NoticeBar, 'NoticeBar');

const { setHTMLProperties, unsetHTMLProperties } = defineHtmlRefProperties({
    offsetWidth: 375,
    offsetHeight: 200,
});

const longText =
    'I became a long long long long long long long long long long long reminder message';

function NoticeBarDemo() {
    const [text, setText] = React.useState('Click I will grow and start scrolling!');
    const barRef = React.useRef();
    return (
        <NoticeBar
            ref={barRef}
            marquee="always"
            autoSetGradientStyle={false}
            onClick={() => {
                setText(longText);
                if (barRef.current) {
                    barRef.current.updateData();
                }
            }}
        >
            {text}
        </NoticeBar>
    );
}

describe('NoticeBar style', () => {
    it('Common styles render correctly', () => {
        const { container } = render(<NoticeBar>Note that this is a reminder message.</NoticeBar>);
        expect(container.textContent).toEqual('Note that this is a reminder message.');
    });

    it('Icon styles render correctly', () => {
        const { container } = render(
            <NoticeBar leftContent={<IconNotice />}>
                Note that this is a reminder message.
            </NoticeBar>,
        );
        const svgArray = container.getElementsByTagName('svg');
        expect(svgArray.length).toBe(2);
    });
});

describe('NoticeBar actions', () => {
    beforeAll(() => {
        // mockContainerSize();
        setHTMLProperties();
    });

    afterAll(() => {
        // resetContainerSizeMock();
        unsetHTMLProperties();
    });

    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('Click render correctly', async () => {
        const onClickRight = jest.fn();
        const { container } = render(
            <div>
                <NoticeBar onClose={onClickRight}>Note that this is a reminder message.</NoticeBar>,
            </div>,
        );
        const closeIcon = container.querySelectorAll(`.${prefixCls}-icon-close`)[0];
        userEvent.click(closeIcon);
        expect(onClickRight.mock.calls.length).toBe(1);
        expect(container.querySelectorAll(`.${prefix}`).length).toBe(0);
    });

    it('Should support using ref to update', () => {
        const { container } = render(<NoticeBarDemo />);
        const noticeBar = container.querySelectorAll(`.${prefix}`)[0];

        userEvent.click(noticeBar);
        pureDelay(2000);
        const content = container.querySelectorAll(`.${prefix}-content-inner`)[0];
        expect(content.textContent).toBe(longText);
    });
});
