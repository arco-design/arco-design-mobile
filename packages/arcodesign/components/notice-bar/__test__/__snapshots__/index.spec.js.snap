// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notice-bar demo test notice-bar demo: custom.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
    style="color: rgb(255, 87, 34); background-color: rgb(255, 241, 232);"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a notification message. The background color is customized here, and the gradient theme color will also be automatically set as the background color changes.
      </div>
    </div>
    <div
      class="arco-notice-bar-right-part"
    >
      <div
        style="background: rgb(255, 87, 34); line-height: 24px; color: white; font-size: 13px; border-radius: 24px; padding: 0px 8px; margin-top: -2px;"
      >
        Turn on
      </div>
    </div>
  </div>
  <div
    class="arco-notice-bar no-wrap"
    style="color: rgb(0, 180, 42); background-color: rgb(232, 255, 234); margin-top: 12px;"
  >
    <div
      class="arco-notice-bar-left-part"
    >
      <svg
        class="arco-icon arco-icon-success-circle "
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 24C5.4 24 0 18.6 0 12S5.4 0 12 0s12 5.4 12 12-5.4 12-12 12zm0-22.5C6.2 1.5 1.5 6.2 1.5 12S6.2 22.5 12 22.5 22.5 17.8 22.5 12 17.8 1.5 12 1.5z"
        />
        <path
          d="M17.5 8c-.3-.3-.7-.3-1 0l-6.3 6.3-3-3c-.3-.3-.7-.3-1 0-.3.3-.3.7 0 1l3.5 3.5c.1.1.3.2.4.2s.4-.1.5-.2L17.5 9c.2-.3.2-.7 0-1z"
        />
      </svg>
    </div>
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        File is uploaded successfully!
      </div>
    </div>
  </div>
  <div
    class="arco-notice-bar no-wrap"
    style="color: rgb(245, 63, 63); background: rgb(255, 236, 232); margin-top: 12px;"
  >
    <div
      class="arco-notice-bar-left-part"
    >
      <svg
        class="arco-icon arco-icon-warn-circle "
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 24C5.4 24 0 18.6 0 12S5.4 0 12 0s12 5.4 12 12-5.4 12-12 12zm0-22.5C6.2 1.5 1.5 6.2 1.5 12S6.2 22.5 12 22.5 22.5 17.8 22.5 12 17.8 1.5 12 1.5z"
        />
        <path
          d="M11.8 15.2h.3c.4 0 .7-.3.7-.7V7.2c0-.4-.3-.7-.7-.7h-.3c-.4 0-.7.3-.7.7v7.3c0 .4.3.7.7.7zM12.2 16.5h-.3c-.4 0-.7.3-.7.7v.3c0 .4.3.7.7.7h.3c.4 0 .7-.3.7-.7v-.3c0-.4-.3-.7-.7-.7z"
        />
      </svg>
    </div>
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        The network connection is not available, please check the network settings.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: ellipsis.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap ellipsis"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message. This message is very long and will be omitted
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: icon.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-left-part"
    >
      <svg
        class="arco-icon arco-icon-notice "
        fill="currentColor"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M618.667 853.333A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248V704h64a21.333 21.333 0 0121.334 21.333V768a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 768v-42.667A21.333 21.333 0 01106.667 704h64V459.008c0-164.181 108.8-302.123 256-341.248V85.333A21.333 21.333 0 01448 64h128zm-64 128c-141.376 0-256 119.595-256 267.136V704h512V459.136C768 311.595 653.376 192 512 192z"
        />
      </svg>
    </div>
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: marquee.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a very long reminder message displayed on a single line, longer than the container width, so it will scroll.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="arco-notice-bar no-wrap"
    style="margin-top: 12px;"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        It's a short message, but also scrollable.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="arco-notice-bar no-wrap"
    style="margin-top: 12px;"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Click me and I'll get longer and start scrolling!
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: more.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message.
      </div>
    </div>
    <div
      class="arco-notice-bar-right-part"
    >
      <svg
        class="arco-icon arco-icon-arrow-in "
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9 8L8.4 2.6c-.1-.1-.1-.3 0-.5l.5-.5c.1-.1.3-.1.5 0l5.9 5.9c.3.3.3.7 0 .9l-5.9 5.9c-.1.1-.3.1-.5 0l-.5-.5c-.1-.1-.1-.3 0-.5L13.9 8z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: none.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: noshow.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        This is a reminder message.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      Do not remind again
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: vertical.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar no-wrap"
  >
    <div
      class="arco-notice-bar-left-part"
    >
      <svg
        class="arco-icon arco-icon-notice "
        fill="currentColor"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M618.667 853.333A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248V704h64a21.333 21.333 0 0121.334 21.333V768a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 768v-42.667A21.333 21.333 0 01106.667 704h64V459.008c0-164.181 108.8-302.123 256-341.248V85.333A21.333 21.333 0 01448 64h128zm-64 128c-141.376 0-256 119.595-256 267.136V704h512V459.136C768 311.595 653.376 192 512 192z"
        />
      </svg>
    </div>
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        <div
          class="arco-carousel-wrap"
        >
          <div
            class="arco-carousel"
          >
            <div
              class="arco-carousel-inner auto vertical"
              style="transition-property: none; transition-duration: 0ms; transform: translateY(0) translateZ(0); width: 100%;"
            >
              <div
                class="arco-carousel-item carousel-item normal-item active vertical"
                style="transform: translateY(0px); padding-top: 0px; padding-bottom: 0px; box-sizing: border-box;"
              >
                <div>
                  Note that this is the first reminder message.
                </div>
              </div>
              <div
                class="arco-carousel-item carousel-item normal-item vertical ssr-float"
                style="transform: translateY(0px); padding-top: 0px; padding-bottom: 0px; box-sizing: border-box;"
              >
                <div>
                  Note that this is the second reminder message.
                </div>
              </div>
              <div
                class="arco-carousel-item carousel-item normal-item vertical ssr-float"
                style="transform: translateY(0px); padding-top: 0px; padding-bottom: 0px; box-sizing: border-box;"
              >
                <div>
                  Note that this is the third reminder message.
                </div>
              </div>
            </div>
          </div>
          <div
            class="arco-carousel-indicator pos-center arco-carousel-indicator-vertical vertical ver-pos-left"
          >
            <i
              class="indicator type-square active"
            />
            <i
              class="indicator type-square"
            />
            <i
              class="indicator type-square"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`notice-bar demo test notice-bar demo: wrap.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-notice-bar wrapable"
  >
    <div
      class="arco-notice-bar-content"
    >
      <div
        class="arco-notice-bar-content-inner"
      >
        Note that this is a reminder message. The message is long, but I can choose to have it scroll or wrap, as it does now.
      </div>
    </div>
    <div
      class="arco-notice-bar-close"
    >
      <svg
        class="arco-icon arco-icon-close "
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
